### PR TITLE
Add upgrade tests

### PR DESCRIPTION
**Pull request by asmigala**

### CODE REVIEW

File: src/test/java/org/bf2/cos/e2e/tests/addon/AddonCrdITCase.java [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 
 import java.time.Duration;
+import java.util.Map;
 
 @Order(0)
+@Tags(value = {
+        @Tag("install"),
+        @Tag("upgrade-phase-1"),
+})
 public class AddonCrdITCase {
```
[the remark this answers is not in the document]
Comment (asmigala) [reply src/test/java/org/bf2/cos/e2e/tests/addon/AddonCrdITCase.java:20]: My thinking was that the metadata would include info about the addon, I see no point in having a copy of the test results in the metadata.

[the remark this answers is not in the document]
Comment (tplevko) [reply src/test/java/org/bf2/cos/e2e/tests/addon/AddonCrdITCase.java:20]: I mean info about the CRD's

[the remark this answers is not in the document]
Comment (asmigala) [reply src/test/java/org/bf2/cos/e2e/tests/addon/AddonCrdITCase.java:20]: I couldn't think of any relevant or useful info about the crds that we should keep. The only thing currently written to the metadata is:
* image for each of the pods
* version and replaces for each of the csvs
as those will change with every addon version.

[the remark this answers is not in the document]
Comment (tplevko) [reply src/test/java/org/bf2/cos/e2e/tests/addon/AddonCrdITCase.java:20]: thanks for explanation

**Pull request merged**